### ISSUE TITLE
Generating conditioned images

### ISSUE THREAD

**Issue by GregHewittDA** [PrajitR/fast-pixel-cnn#1]
I want to commend the team on their fantastic work. This addition to the PixelCNN++ makes it much more usable for integrators like myself.

I've been attempting to use your package to produce conditioned images. That is, I used the CIFAR-10 pre-trained model available on PixelCNN++, and passed a one-hot vector to 'h' in your model. I receive the following error:

NotFoundError (see above for traceback): Tensor name "model/conditional_weights_11/hw/ExponentialMovingAverage" not found in checkpoint files /home/da/tensorflow3/pixel-cnn/params_cifar.ckpt

Is this because the pretrained model provided by PixelCNN++ was not trained with the conditional labels? I can see you're restoring variables using variables_to_restore around like 70 of generate.py, but the behavior of this function is complex. As an example, if I restore all variables instead of using the method you employ, the model cannot load v_stack_cache.

Your advice is greatly appreciated.

**Comment by PrajitR** [comment]
Yes, unfortunately OpenAI only released a pretrained model with unconditional generation. If you want conditional generation, you need to train your own model. The good news is that you should be able to use OpenAI's open source training code without modification.

**Comment by GregHewittDA** [comment]
@PrajitR I just wanted to follow up and indicate I used PixelCNN++ with the MNIST dataset with the conditioning in the hidden layer. Your code did a fine job of generating digits of my selection. Thank you.

**Comment by gaceladri** [comment]
Hello guys,

Thank you so much for your great work @PrajitR. I am trying to train the model with different sets and also if it is possible also conditioned. @GregHewittDA How did you adapted the model? Are you going to upload it to anywhere?

The model from pixelcnn++ is as follow:

`def model_spec(x, h=None, init=False, ema=None, dropout_p=0.5, nr_resnet=5, nr_filters=160, nr_logistic_mix=10, resnet_nonlinearity='concat_elu'):`

Where the model from fast_pixelcnn++ is:

`def model_spec(row_input, pixel_input, row, col, image_size, h=None, ema=None, dropout_p=0.5, nr_resnet=5, nr_filters=160, nr_logistic_mix=10, resnet_nonlinearity='concat_elu', seed=None):`

Any insights on this? What row_input, pixel_input, row, col means? row and cols it's clear, but I am a bit confused with this.

Thanks in advance.
